examples/chatgpt/sharepoint_azure_function/Using_Azure_Functions_and_Microsoft_Graph_to_Query_SharePoint.md

```diff
@@ -0,0 +1,951 @@
+   const bearerToken = req.headers.authorization.split(' ')[1];
+   let accessToken;
+   try {
+       accessToken = await getOboToken(bearerToken);
+   } catch (error) {
+       context.res = {
+           status: 500,
+           body: `Failed to obtain OBO token: ${error.message}`
+       };
+       return;
+   }
+   // Initialize the Graph Client using the initGraphClient function defined above
+   let client = initGraphClient(accessToken);
+   // this is the search body to be used in the Microsft Graph Search API: https://learn.microsoft.com/en-us/graph/search-concept-files
+   const requestBody = {
+       requests: [
+           {
+               entityTypes: ['driveItem'],
+               query: {
+                   queryString: searchTerm
+               },
+               from: 0,
+               // the below is set to summarize the top 10 search results from the Graph API, but can configure based on your documents.
+               size: 10
+           }
+       ]
+   };
+
+
+   try {
+       // This is where we are doing the search
+       const list = await client.api('/search/query').post(requestBody);
+       const processList = async () => {
+           // This will go through and for each search response, grab the contents of the file and summarize with gpt-3.5-turbo
+           const results = [];
+           await Promise.all(list.value[0].hitsContainers.map(async (container) => {
+               for (const hit of container.hits) {
+                   if (hit.resource["@odata.type"] === "#microsoft.graph.driveItem") {
+                       const { name, id } = hit.resource;
+                       // The below is where the file lives
+                       const driveId = hit.resource.parentReference.driveId;
+                       // we use the helper function we defined above to get the contents, convert to base64, and restructure it
+                       const contents = await getDriveItemContent(client, driveId, id, name);
+                       results.push(contents)
+               }
+           }));
+           return results;
+       };
+       let results;
+       if (list.value[0].hitsContainers[0].total == 0) {
+           // Return no results found to the API if the Microsoft Graph API returns no results
+           results = 'No results found';
+       } else {
+           // If the Microsoft Graph API does return results, then run processList to iterate through.
+           results = await processList();
+           // this is where we structure the response so ChatGPT knows they are files
+           results = {'openaiFileResponse': results}
+       }
+       context.res = {
+           status: 200,
+           body: results
+       };
+   } catch (error) {
+       context.res = {
+           status: 500,
+           body: `Error performing search or processing results: ${error.message}`,
+       };
+   }
+};
+```
+### Solution 1 Customizations
```

Comment:
nit: Instead of having "Solution 1 [section title]" just use [section title], and go down one heading if necessary